Result/DocumentScanIterator.php

```diff
@@ -130,4 +130,29 @@ public function next()
     {
         $this->key++;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetGet($offset)
+    {
+        if (!isset($this->converted[$offset])) {
+            if (!isset($this->documents[$offset])) {
+                return null;
+            }
+
+            $this->converted[$offset] = $this->convertDocument($this->documents[$offset]);
+
+            // Clear memory.
+            if (isset($this->converted[$offset - 20])) {
```

Comment:
@tfe2012 should know the exact reason, but I believe this was a hack. Integer `1` or even `2` should be enough, because you cannot iterate the scan iterator again anyways.